src/printer_tree.c

```diff
@@ -4337,3 +4418,131 @@ tree_print_parsed_submodule(struct ly_out *out, const struct lysp_submodule *sub
+        struct lyd_value_xpath10 *xp_val;
+        char *value;
+        struct ly_err_item *err;
+
+        term = (struct lyd_node_term *)pref_set->dnodes[i];
+        LYD_VALUE_GET(&term->value, xp_val);
+        LY_CHECK_GOTO(res = lyplg_type_print_xpath10_value(xp_val, LY_VALUE_JSON, NULL, &value, &err), out);
+        LY_CHECK_ERR_GOTO(res = lys_find_xpath(ext->module->ctx, NULL, value, 0, &snode_set), free(value), out);
+        free(value);
+        for (uint32_t sn = 0; sn < snode_set->count; sn++) {
+            struct lysc_node *top = snode_set->snodes[sn];
+
+            while (top->parent)
+                top = top->parent;
+
+            if ((res = ly_set_add(results_set, top, 0, NULL))) {
+                ly_set_free(snode_set, NULL);
+                ly_set_free(results_set, NULL);
+                goto out;
+            }
+        }
+        ly_set_free(snode_set, NULL);
+    }
+
+    *refs = results_set;
+
+out:
+    if (ext_data_free) {
+        lyd_free_all(ext_data);
+    }
+    ly_set_free(pref_set, NULL);
+    return res;
+}
+
+static LY_ERR
+trb_print_mount_point(struct trt_node *node, struct trt_wrapper wr, struct trt_printer_ctx *pc,
+                      struct trt_tree_ctx *UNUSED(tc))
+{
+    struct ly_ctx *ext_ctx;
+    const struct lys_module *mod;
+    struct trt_tree_ctx tmptc;
+    struct trt_tree_ctx *newtc;
+    struct trt_wrapper tmpwr;
+    struct ly_set *modules;
+    struct ly_set *refs = NULL;
+    LY_ERR err = LY_SUCCESS;
+
+    if (lyplg_ext_schema_mount_create_context(node->mount, &ext_ctx))
+        /* Void mount point */
+        return LY_SUCCESS;
+
+    if (ly_set_new(&modules)) {
+        err = LY_SUCCESS;
+        goto out_ctx;
+    }
+
+    trb_parent_references(node->mount, &refs);
+
+    /* build new list of modules to print.  This list will omit internal
+     * modules and modules with no nodes (e.g., iana-if-types).
+     */
+    uint32_t v = ly_ctx_internal_modules_count(ext_ctx);
+    while ((mod = ly_ctx_get_module_iter(ext_ctx, &v))) {
+        uint32_t siblings;
+
+        if ((mod->ctx->flags & LY_CTX_SET_PRIV_PARSED) && mod->compiled)
+            trm_lysc_tree_ctx(mod, pc->out, pc->max_line_length, pc, &tmptc);
+        else
+            trm_lysp_tree_ctx(mod, pc->out, pc->max_line_length, pc, &tmptc);
+
+        siblings = (tmptc.pn || tmptc.cn)
+                   ? trb_get_number_of_siblings(pc->fp.modify, &tmptc) : 0;
+        if (siblings == 0)
+            continue;
+        tmptc.mounted = 1;
+        tmptc.parent_refs = refs;
+
+        newtc = malloc(sizeof *newtc);
+        if (newtc == NULL) {
+            err = LY_EMEM;
+            goto out;
+        }
+
+        *newtc = tmptc;
+        ly_set_add(modules, (void *)newtc, 1, NULL);
+    }
+
+    for (uint32_t v = 0; v < modules->count; ++v) {
+        newtc = modules->objs[v];
+        tmpwr = (v == modules->count-1) ? wr : trp_wrapper_set_mark_top(wr);
+        trb_print_family_tree(tmpwr, pc, newtc);
```

Comment:
As I mentioned in a previous comment, the entire subtree should not be printed. The  `trb_print_parents` would be maybe better to use but it will need to be modified a bit. Unfortunately, such a function is missing for a module that is only parsed and not compiled and must be added.